Samples/excel-data-types-explorer/src/taskpane/taskpane.ts

```diff
@@ -0,0 +1,1023 @@
+  switch (valueType) {
+    case "String":
+      return stringContent();
+    case "Double":
+      return doubleContent();
+    case "Boolean":
+      return booleanContent();
+    case "FormattedNumber":
+      return formattedNumberContent();
+    case "WebImage":
+      return webImageContent();
+    case "Entity":
+      return entityContent();
+    case "Unsupported":
+    default:
+      return unsupportedContent();
+  }
+}
+
+/** Create the HTML for entity contents section. */
+function entityContents(valueType: string): JQuery<HTMLElement> {
+  var trKey = $(`<tr>
+          <td><label class="labels">Key:</label></td>
+          <td><input class="inputBox fieldName" alt="Key input box"/></td>
+        </tr>`);
+
+  var tdMetadata = $(`<td colspan="4" class="center settings"/>`);
+
+  if (valueType == "WebImage") {
+    tdMetadata.append(checkboxWithLabel("mainImage", "Make main image", "main image checkbox"));
+    tdMetadata.append($("<br/>"));
+  }
+
+  tdMetadata.append([
+    checkboxWithLabel("cardView", "Card View", "cardview checkbox"),
+    checkboxWithLabel("autoComplete", "Autocomplete", "autoComplete checkbox"),
+    $("<br/>"),
+    checkboxWithLabel("calcCompare", "Calc compare", "calcCompare checkbox"),
+    checkboxWithLabel("dotNotation", "Dot notation", "dotNotation checkbox"),
+    $("<br/>"),
+    $("<label/>").text("Sublabel: "),
+    $("<input/>", { class: "sublabel", alt: "sublabel input box" })
+  ]);
+
+  var trMetadata = $(`<tr class="metadata" style="visibility:collapse"/>`).append(tdMetadata);
+
+  var trButtons = $(`<tr/>`);
+  var tdButtons = $(`<td colspan="4"/>`);
+
+  var buttonToggleMetadata = $(
+    `<button class="ms-Button ms-Button-label buttons" alt="Toggle to expand or collapse metadata properties of field">More settings</button>`
+  );
+
+  /** Expand or collapse the additional metadata contents of a
+  particular input field, within the entity contents. */
+  buttonToggleMetadata[0].onclick = function() {
+    var visibility = trMetadata[0].style.visibility;
+    if (visibility != "collapse") {
+      trMetadata[0].style.visibility = "collapse";
+    } else {
+      trMetadata[0].style.visibility = "visible";
+    }
+  };
+
+  tdButtons.append([buttonToggleMetadata]);
+  trButtons.append(tdButtons);
+
+  return $(`<div id="entityContents"/>`).append(
+    $(`<table id="fieldTable"/>`).append(
+      $(`<tbody/>`)
+        .append(trKey)
+        .append(getTypeContent(valueType))
+        .append([trMetadata, trButtons])
+    )
+  );
+}
+
+/** Create the HTML for the entity data type. */
+function entityContent(): JQuery<HTMLElement> {
+  var div = $(`<div class="contentPadding"/>`);
+  var iconlabel = $(`<label class="labels"> Entity icon: </label>`);
+  var contentLabel = $(`<label class="labels contentPadding">Entity contents:</label>`);
+  var select = $(`<select id="iconSelect" name="dataType" class="ms-Button ms-Button-label buttons"/>`);
+  var options = createOptionsFromList([
```

Comment:
I think you can generate this list dynamically using `Object.values(MyEnum)`.

Reply:
![image](https://user-images.githubusercontent.com/38896772/190501596-36dac482-4753-44ea-95a4-3645d424e1ad.png)

Not sure. Are you getting that same error with an old enum (that's not in preview)?

Reply:
I'm only getting this error with EntityCompactLayoutIcons. Other enums work for me. I'm consulting eng about it — it could be a discrepancy between Windows/Mac. For now, I think we should keep the static icon list here.